### DISCUSSION TITLE
Intelligent RSS download

### DISCUSSION THREAD

**Discussion by Aohzan** [jesec/flood#83] [Ideas]
Type: Feature Request

First, @jesec, thanks for your work 

## Summary
When auto download from RSS feeds, don’t take multiple time, the same episode number.

## Idea of implementation
From regex rules, detect episode number like s01e01 or 1x1

**Comment by jesec** [comment]
Flood won’t requeue the item if an item with the same URL has already been queued.

Won’t there be other differences (like resolution, publisher, etc) in title even if the episode number is the same? It seems unlikely to me that there will be feed items that are exactly same in title.

Current RegEx based matches should be sufficient for all those cases. Admittedly you might have to use a more sophisticated expression. Testers like [regex101.com](https://regex101.com) can help here.

**Comment by Aohzan** [comment]
A single feed can have multiple release from the same episode number (example: title.s01e01.720p.hdtv and title.s01e01.1080.bdrip...).
It's in this case that Flood will download every release of I just put `title.*` in the search pattern. 

Currently, to avoid this, my search pattern have a fixed resolution, fixed source, fixed team. But if the specific resolution set isn't in the feed, I can miss a episode

**Comment by Aohzan** [comment]
uTorrent have the feature :
![rss_demo](https://user-images.githubusercontent.com/2736322/98782880-0608c900-23f9-11eb-91bd-b6555084d650.png)

**Comment by jesec** [comment]
It is expected that rules need to have precise patterns. Regular expressions support OR. You may specify multiple resolution strings to match for example.

**Comment by Aohzan** [comment]
Yes I know, but it will download all the resolution as soon as they are in the feed. In my case, I only want the first release available for the episode

**Comment by jesec** [comment]
Um... I think in most cases, there are clear preferences in certain properties like resolution or publisher. It is a rare case that the time of availability go above these preferences. 

Even if there is such need, most people probably won't be concerned about the duplicates. Disk and bandwidth are cheap these days (and it won't be particularly burdensome to remove duplicates). 

Some new record keeping functions would have to be implemented for this feature.

**Comment by Aohzan** [comment]
Ok I understand the complexity to implement this feature, not usefull for everyone. Thanks
